Sources/MessagingPush/RichPush/MessagingPush+RichPush.swift

```diff
@@ -73,30 +73,19 @@ extension MessagingPushImplementation {
 
             // This conditional will only work in production and not in automated tests. But this file cannot be in automated tests so this conditional is OK for now.
             if let composedRichPush = composedRichPush as? UNNotificationWrapper {
-                self.finishTasksThenReturn(contentHandler: contentHandler, notificationContent: composedRichPush.notificationContent)
+                self.logger
+                    .debug(
+                        "running all background queue tasks and waiting until complete to prevent OS from killing notification service extension before all HTTP requests have been performed"
+                    )
+
+                self.logger.info("Customer.io push processing is done!")
+                contentHandler(composedRichPush.notificationContent)
```

Comment:
When we call the `contentHandler()`, we are telling the OS that our NSE is completely done doing what it needs to do. 

This might be out of scope of this PR, but I think that we need to make sure that before we call the contentHandler, we confirm the `delivered` metric HTTP request has finished, correct? 

I see above this line of code `trackMetricFromNSE(deliveryID: pushCioDeliveryInfo.id, event: .delivered, deviceToken: pushCioDeliveryInfo.token)` and because there is no async callback in that line of code, we are not informed when the HTTP request concludes, correct?

Reply:
yup, that is correct, i'll create a separate PR for it. But good callout.